test/distributed/_sharded_optim/test_sharded_optim.py

```diff
@@ -105,6 +146,40 @@ def test_sharded_optim(self):
                 self.assertNotEqual(val, new_val)
                 self.assertEqual(new_val, local_model.param)
 
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_named_params_with_sharded_tensor(self):
+        rowwise_spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+            ],
+        )
+        sharded_model = MyShardedModel(spec=rowwise_spec).cuda(self.rank)
+        sharded_model_params = dict(named_params_with_sharded_tensor(sharded_model))
+        param_keys = list(sharded_model_params.keys())
+        self.assertEqual(len(param_keys), 2)
+        self.assertTrue("param" in param_keys)
+        self.assertTrue("sharded_param" in param_keys)
+
+        sharded_linear = MyShardedLinear(rank=self.rank).cuda(self.rank)
+        sharded_linear.shard_parameter()
+        sharded_linear_params = dict(named_params_with_sharded_tensor(sharded_linear))
+        param_keys = list(sharded_linear_params.keys())
+        self.assertEqual(len(param_keys), 4)
+        self.assertTrue("linear1.bias" in param_keys)
+        self.assertTrue("linear2.bias" in param_keys)
+        self.assertTrue("linear1.weight" in param_keys)
+        self.assertTrue("linear2.weight" in param_keys)
+        self.assertFalse("bias" in param_keys)
```

Comment:
Would this unit test fail without the fix below?

Reply:
yep tested locally this test will fail without the fix, it will have an additional "bias" param recorded.